pytensor/tensor/pad.py

```diff
@@ -0,0 +1,536 @@
+        outputs_info=[0, None],
+        n_steps=repeats,
+    )
+
+    parts = moveaxis(parts, 0, axis)
+    new_shape = [-1 if i == axis else array.shape[i] for i in range(array.ndim)]
+    padding = parts.reshape(new_shape)
+
+    return padding
+
+
+def _symmetric_pad(x, pad_width):
+    def _symmetric_inner(i, x, x_flipped, padding_left):
+        return i + 1, ifelse(eq(i % 2, int(padding_left)), x_flipped, x)
+
+    pad_width = broadcast_to(pad_width, as_tensor((x.ndim, 2)))
+
+    for axis in range(x.ndim):
+        x_flipped = flip(x, axis=axis)
+        original_size = x.shape[axis]
+
+        repeats, remainders = pt_divmod(pad_width[axis], original_size)
+        has_remainder = gt(remainders, 0)
+        repeats = repeats + has_remainder
+
+        left_padding = _build_padding_one_direction(
+            x,
+            x_flipped,
+            repeats[0],
+            axis=axis,
+            inner_func=partial(_symmetric_inner, padding_left=True),
+        )
+        right_padding = _build_padding_one_direction(
+            x,
+            x_flipped,
+            repeats[1],
+            axis=axis,
+            inner_func=partial(_symmetric_inner, padding_left=False),
+        )
+
+        x = concatenate([flip(left_padding, axis), x, right_padding], axis=axis)
+
+        (left_trim, right_trim) = switch(
+            has_remainder, original_size - remainders, remainders
+        )
+        right_trim = x.shape[axis] - right_trim
+
+        trim_slice = _slice_at_axis(slice(left_trim, right_trim), axis)
+        x = x[trim_slice]
+
+    return x
+
+
+def _reflect_pad(x, pad_width):
+    def _reflect_inner(i, x, x_flipped, padding_left):
+        return i + 1, ifelse(eq(i % 2, int(padding_left)), x_flipped, x)
+
+    pad_width = broadcast_to(pad_width, as_tensor((x.ndim, 2)))
+    for axis in range(x.ndim):
+        trimmed_size = x.shape[axis] - 1
+
+        trim_slice = _slice_at_axis(slice(None, -1), axis)
+        x_trimmed = x[trim_slice]
+        x_flipped = flip(x, axis=axis)[trim_slice]
+
+        repeats, remainders = pt_divmod(pad_width[axis], trimmed_size)
+        repeats = repeats + 1
+
+        left_padding = _build_padding_one_direction(
+            x_trimmed,
+            x_flipped,
+            repeats[0],
+            axis=axis,
+            inner_func=partial(_reflect_inner, padding_left=True),
+        )
+        right_padding = _build_padding_one_direction(
+            x_trimmed,
+            x_flipped,
+            repeats[1],
+            axis=axis,
+            inner_func=partial(_reflect_inner, padding_left=False),
+        )
+
+        left_trim = _slice_at_axis(slice(trimmed_size - remainders[0] - 1, -1), axis)
+        right_trim = _slice_at_axis(
+            slice(1, right_padding.shape[axis] - trimmed_size + remainders[1] + 1), axis
+        )
+
+        x = concatenate(
+            [flip(left_padding, axis)[left_trim], x, right_padding[right_trim]],
+            axis=axis,
+        )
+    return x
```

Comment:
Do we need OpFromGraph? Everytime we hide it we tradeoff spotting it at the expensive of reasoning about how its lower level components / interacting with the outer graphs

Reply:
No, I suppose not. I like it because it makes the `dprint` much easier to understand. But if there are actual performance costs, we can remove it.

Reply:
I don't know if there are perf issues, there may be redundant computation if you have two pads in the same graph that share some steps. 

I would only do OFG when we find a perf reason for it (need custom grad, need to target in rewrites)

Reply:
You can also make it inline by default and then you can exclude inline in the mode if you want to keep a readable graph foe your workflow?